src/lib/dnssd/minimal_mdns/AddressPolicy_LibNlImpl.cpp

```diff
@@ -176,14 +182,42 @@ bool AllListenIterator::Next(InterfaceId * id, IPAddressType * type)
             continue;
         }
 
-        int idx = rtnl_link_get_ifindex(CurrentLink());
-        if (idx == 0)
+        int ifindex = rtnl_link_get_ifindex(CurrentLink());
+        if (ifindex == 0)
         {
             // Invalid index, move to the next interface
             continue;
         }
 
-        *id   = InterfaceId(idx);
+        // For IPv4, report only interfaces which have an IPv4 address
```

Comment:
Will this make it so that an interface with only an IPv6 address will be skipped for v6 as well, just because IPv4 is enabled?

Reply:
No. The code always checks IPv6 then IPv4 before moving to the next link. This change only affects IPv4.

I could add ifdefs here, but it should not matter in practice.